### ISSUE TITLE
Where clauses from error types are not propagated to implementations

### ISSUE THREAD

**Issue by HeapUnderfl0w** [shepmaster/snafu#53]
Given this code:
```rust
use crate::math::Op;
use std::fmt::Debug;

#[derive(Debug, PartialEq, Snafu)]
pub enum EvalError<T> where T: Debug {
    #[snafu(display("overflow occured while evaluating \"{:?} {} {:?}\"", left, op, right))]
    Overflow { left: T, op: Op, right: T },
    #[snafu(display("unknown error"))]
    Unknown
}
```
with op being defined as 
```rust
#[derive(Debug, Clone, Copy, Eq, PartialEq)]
pub enum Op {
    Add,
    Sub,
    Mul,
    Div,
    Pow,
}

impl fmt::Display for Op {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        write!(
            f,
            "{}",
            match self {
                Op::Add => "+",
                Op::Sub => "-",
                Op::Mul => "*",
                Op::Div => "/",
                Op::Pow => "^",
            }
        )
    }
}
```
rustc emits a missing `std::fmt::Debug` implementation, even though the type clearly has a generic constraint for exactly that.  

```
error[E0277]: `T` doesn't implement `std::fmt::Debug`
  --> src/error.rs:23:28
   |
23 | #[derive(Debug, PartialEq, Snafu)]
   |                            ^^^^^ `T` cannot be formatted using `{:?}` because it doesn't implement `std::fmt::Debug`
   |
   = help: the trait `std::fmt::Debug` is not implemented for `T`
   = help: consider adding a `where T: std::fmt::Debug` bound
note: required by `error::EvalError`
  --> src/error.rs:24:1
   |
24 | pub enum EvalError<T> where T: Debug {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to previous error
```

Removing snafu from the enum makes it compile as expected.

Rustc: `rustc 1.35.0-nightly (3de010678 2019-04-11)`  
Cargo: `cargo 1.35.0-nightly (6f3e9c367 2019-04-04)`
Snafu: `0.2.2`

**Comment by shepmaster** [comment]
Reduced test case:

```rust
use snafu::Snafu;
use std::fmt::Debug;

#[derive(Debug, Snafu)]
pub enum Error<T>
where
    T: Debug,
{
    #[snafu(display("Inner value is {:?}", inner))]
    Alpha { inner: T },
}
```

**Comment by shepmaster** [comment]
Thanks for the report!

As a heads-up, you have a typo ("occurred") in your message. You can also use raw strings to avoid the escaping of quotes:

```rust
#[snafu(display(r#"overflow occurred while evaluating "{:?} {} {:?}"#, left, op, right))]
```

**Comment by shepmaster** [comment]
Released in 0.2.3; thanks for using SNAFU!
